src/entities/bond.ts

```diff
@@ -94,6 +101,14 @@ export class Bond {
             'Invalid input currency - not bond collateral',
         );
 
+        invariant(
+            this.depositLimit.eq(0) ||
+                this.totalCollateral
+                    .add(collateralInput.quotient.toString())
```

Comment:
is collateralInput scaled by decimals? I'm assuming totalCollateral is since it's a BigNumber, and asking because quotient is used on collateralInput